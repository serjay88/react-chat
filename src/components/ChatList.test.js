/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatList from './ChatList';

jest.mock('./ChatListItem', () => ({ chatId }) => `ChatListItem-${chatId}`);

const mockData = {
  chats: [
    {
      _id: '12345',
      title: 'My Chat',
      createdAt: '2018-03-29T20:32:34.303Z',
    },
    {
      _id: '34567',
      title: 'Other chat',
      createdAt: '2018-03-29T20:32:34.303Z',
    },
  ],
  activeChat: {
    _id: '12345',
  },
  disabled: false,
};

describe('<ChatList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatList {...mockData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly ', () => {
    const tree = renderer.create(<ChatList {...mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
