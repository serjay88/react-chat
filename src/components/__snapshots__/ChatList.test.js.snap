// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly  1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatList-chatsList-7"
>
  ChatListItem-12345
  ChatListItem-34567
</ul>
`;
